### DISCUSSION TITLE
Enable (simple) Routines

### DISCUSSION THREAD

**Discussion by tomrennen** [hass-emulated-hue/core#103] [Ideas]
### Solution: Generate HA automations based on Hue App input. 

This could enable the use of time based routines through the Hue app, by generating an automation that is triggered on said time and adding `transition` to `light.turn_on`. This would only support lights that support `transition` though, so there might be a better alternative (some blueprints already do this e.g. https://community.home-assistant.io/t/wake-up-light-alarm-with-sunrise-effect/255193).

I was looking at this discussion: https://community.home-assistant.io/t/rest-api-docs-for-automations/119997/10 . It seems fairly straight forward, "just" generate json according to the automation. This does involve keeping track of generated automations or their id's I suppose, as you might want to delete/edit existing automations. Not sure how convenient the HA API is in this regard.

**Comment by alexyao2015** [comment]
This is a great idea, however we would first need to implement #50 as it's directly related to this and squash some other bugs before being able to move forward with implementing this.

**Reply by alexyao2015** [reply]
Perhaps, would you be able to put your comments here in #50, so we have it on the roadmap for when that does get implemented?
